### ISSUE TITLE
Various Passenger tools don't work on Solaris 11.2 (problems with flock)

### ISSUE THREAD

**Issue by CirrusThink** [phusion/passenger#1643]
same error when rvmsudo sudo or just passenger-status

~~~
/home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/lib/phusion_passenger/admin_tools/instance.rb:48:in `flock': Bad file number @ rb_file_flock - /tmp/passenger.ZDMYRxk/lock (Errno::EBADF)
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/lib/phusion_passenger/admin_tools/instance.rb:48:in `block in locked?'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/lib/phusion_passenger/admin_tools/instance.rb:47:in `open'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/lib/phusion_passenger/admin_tools/instance.rb:47:in `locked?'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/lib/phusion_passenger/admin_tools/instance_registry.rb:32:in `block (2 levels) in list'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/lib/phusion_passenger/admin_tools/instance_registry.rb:28:in `each'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/lib/phusion_passenger/admin_tools/instance_registry.rb:28:in `block in list'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/lib/phusion_passenger/admin_tools/instance_registry.rb:27:in `each'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/lib/phusion_passenger/admin_tools/instance_registry.rb:27:in `list'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/bin/passenger-status:47:in `find_sole_instance'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/bin/passenger-status:39:in `command_show_status'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/bin/passenger-status:320:in `start'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/gems/passenger-enterprise-server-5.0.15/bin/passenger-status:323:in `<top (required)>'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/bin/passenger-status:23:in `load'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/bin/passenger-status:23:in `<main>'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/bin/ruby_executable_hooks:15:in `eval'
        from /home/dolly/.rvm/gems/ruby-2.1.5@global/bin/ruby_executable_hooks:15:in `<main>'
~~~

The app seems to be running but cat query with passenger-status

**Comment by FooBarWidget** [comment]
It looks like Solaris is trying to emulate the flock() function with fcntl() locks, but for some reason it fails. Deeper investigation is needed.

**Comment by FooBarWidget** [comment]
I was counting on Ruby not providing the flock function on Solaris at all. But apparently Ruby does provide it, it just provides a broken version. I'll have to find a way to work around this....

**Comment by FooBarWidget** [comment]
The issue has been fixed now. The fix will be part of 5.0.22, scheduled for Monday.
